examples/agents/airTableToolAgent.ts

```diff
@@ -0,0 +1,157 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import "dotenv/config.js";
+import { BAMChatLLM } from "bee-agent-framework/adapters/bam/chat";
+import { BeeAgent } from "bee-agent-framework/agents/bee/agent";
+import { createConsoleReader } from "../helpers/io.js";
+import { FrameworkError } from "bee-agent-framework/errors";
+import { TokenMemory } from "bee-agent-framework/memory/tokenMemory";
+import { Logger } from "bee-agent-framework/logger/logger";
+import { OllamaChatLLM } from "bee-agent-framework/adapters/ollama/chat";
+import { OpenAIChatLLM } from "bee-agent-framework/adapters/openai/chat";
+import { PromptTemplate } from "bee-agent-framework/template";
+import { WatsonXChatLLM } from "bee-agent-framework/adapters/watsonx/chat";
+import { parseEnv } from "bee-agent-framework/internals/env";
+import { z } from "zod";
+import {
+  BeeSystemPrompt,
+  BeeToolErrorPrompt,
+  BeeToolInputErrorPrompt,
+  BeeToolNoResultsPrompt,
+} from "bee-agent-framework/agents/bee/prompts";
+
+// core tools
+import { DuckDuckGoSearchTool } from "bee-agent-framework/tools/search/duckDuckGoSearch";
+import { WikipediaTool } from "bee-agent-framework/tools/search/wikipedia";
+
+// AirTable tool
+import { AirtableTool } from "bee-community-tools/tools/airtable";
+
+const AIRTABLE_TOKEN: string = parseEnv("AIRTABLE_TOKEN", z.string());
+const AIRTABLE_BASE: string = parseEnv("AIRTABLE_BASE", z.string());
+
+Logger.root.level = "silent"; // disable internal logs
+const logger = new Logger({ name: "app", level: "trace" });
+
+async function runBeeAgent() {
+  // use BAM if GENAI_API_KEY env var is defined
+  // else use Watsonx if WATSONX_API_KEY and WATSONX_PROJECT_ID env vars are defined
+  // else use OpenAI if OPENAI_API_KEY env var is defined
+  // else use Ollama
+  const llm =
+    process.env.GENAI_API_KEY !== undefined
+      ? BAMChatLLM.fromPreset("meta-llama/llama-3-1-70b-instruct")
+      : process.env.WATSONX_API_KEY !== undefined && process.env.WATSONX_PROJECT_ID !== undefined
+        ? WatsonXChatLLM.fromPreset("meta-llama/llama-3-1-70b-instruct", {
+            apiKey: process.env.WATSONX_API_KEY, //pragma: allowlist secret
+            projectId: process.env.WATSONX_PROJECT_ID,
+            parameters: {
+              decoding_method: "greedy",
+              min_new_tokens: 5,
+              max_new_tokens: 50,
+            },
+          })
+        : process.env.OPENAI_API_KEY !== undefined
+          ? new OpenAIChatLLM()
+          : new OllamaChatLLM({ modelId: "llama3.1" });
```

Comment:
Consider using a factory function.

Reply:
Yeah, this was a bit of a quick-and-dirty hack!  Ternaries are bad enough but nested ternaries make for horrible reading, hence the comment above the code.